diff -ur aterm-0.4.2.orig/src/feature.h aterm-0.4.2/src/feature.h
--- aterm-0.4.2.orig/src/feature.h	2004-08-12 16:19:57.518239976 +0200
+++ aterm-0.4.2/src/feature.h	2004-08-12 16:20:05.732991144 +0200
@@ -123,7 +123,7 @@
  * instead of
  *	Home = "\E[7~", End = "\E[8~"	[default]
  */
-#define LINUX_KEYS
+//#define LINUX_KEYS
 
 /*
  * Enable the keysym resource which allows you to define strings associated
